### PR TITLE
'Multi_Threading works -> seem slower for now. Need to add for multi …

### PR DESCRIPTION
**Pull request by tchaton**
…radiuses'

### CODE REVIEW

**Comment by tchaton** [comment]
Normal multi processing working. I will add multi radiuses when current PRs are merged

**Comment by tchaton** [comment]
On cpu,
Time spent in Fps | Radius Search

0.007954120635986328 0.22580170631408691
0.00791788101196289 0.22681164741516113
0.0077555179595947266 0.22717928886413574
0.007649660110473633 0.22800397872924805
0.0019350051879882812 0.013108253479003906
0.001928567886352539 0.013298273086547852
0.0019190311431884766 0.013692140579223633
0.0018720626831054688 0.013240814208984375
0.006800413131713867 0.22521257400512695
0.006579160690307617 0.2246840000152588
0.006214618682861328 0.22668886184692383
0.006578922271728516 0.2267296314239502
0.001828908920288086 0.012511253356933594
0.0017671585083007812 0.012578964233398438
0.0017476081848144531 0.013388872146606445
0.001567840576171875 0.013301610946655273
0.00690460205078125 0.22823548316955566
0.007228374481201172 0.228285551071167
0.007268190383911133 0.2281484603881836
0.007357120513916016 0.2292342185974121
0.001760244369506836 0.012793302536010742
0.0023126602172851562 0.012842893600463867
0.0016376972198486328 0.013065338134765625
0.0016634464263916016 0.013552427291870117
0.00728154182434082 0.2388622760772705

File: datasets/transforms.py [hunk 1]
```diff
@@ -37,20 +40,44 @@ def __init__(self, strategies, precompute_multi_scale=False):
             raise Exception('Strategies are empty and precompute_multi_scale is set to True')
         self.num_layers = len(self.strategies.keys())
 
-    def __call__(self, data):
+    @staticmethod
+    def __inc__wrapper(func, special_params):
+        def new__inc__(key, num_nodes, special_params=None, func=None):
```
Comment (nicolas-chaulet) [inline datasets/transforms.py]: func is never used it seems

File: datasets/transforms.py [hunk 2]
```diff
@@ -37,20 +40,44 @@ def __init__(self, strategies, precompute_multi_scale=False):
...
+                num_nodes_for_edge_index = torch.from_numpy(np.array([pos.shape[0],
+                                                                      pos[idx].shape[0]])).unsqueeze(-1)
+
+                special_params[index_name] = num_nodes_for_edge_index[0]
+
+                special_params[edge_name] = num_nodes_for_edge_index
                 pos = pos[idx]
                 batch = batch[idx]
+            func_ = self.__inc__wrapper(data.__inc__, special_params)
+            setattr(data, "__inc__", func_)
```
Comment (nicolas-chaulet) [inline datasets/transforms.py]: It looks a bit risky to override a private method like that. Are you 100% sure you want to do that?

File: datasets/base_dataset.py [hunk 3]
```diff
@@ -59,14 +48,19 @@ def weight_classes(self):
...
         for _, attr in self.__dict__.items():
             if isinstance(attr, DataLoader):
-                def collate_fn(data_list): return BatchWithTransform.from_data_list_with_transform(
-                    data_list, [], batch_transform)
-                setattr(attr, "collate_fn", collate_fn)
+                current_transform = getattr(attr.dataset, "transform", None)
+                if current_transform is None:
+                    setattr(attr.dataset, "transform", transform)
+                else:
+                    if isinstance(current_transform, T.Compose):  # The transform contains several transformations
```
Comment (nicolas-chaulet) [inline datasets/base_dataset.py]: Are you sure you cannot treat a composed transform just like a normal one?

**Pull request merged**